bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/ExecutorImpl.scala

```diff
@@ -13,39 +16,56 @@ import scalaz.concurrent.Task
 /** An executor that maintains state in a pair of `TaskRef`s. */
 class ExecutorImpl private (cancelRef: TaskRef[Set[SPObservationID]], stateRef: TaskRef[Map[SPObservationID, Executor.ExecState]]) {
 
+  private var loc = new Peer("localhost", 8443, null)
+
+  def host(): Peer = loc
+  def host(l: Peer): Unit = { loc = l }
+
   private def recordState(id: SPObservationID)(s: ExecState): Task[Unit] =
     stateRef.modify(_ + (id -> s))
 
   private def go(id: SPObservationID): Task[Boolean] = 
     cancelRef.get.map(!_(id))
 
-  def sequence(sequenceConfig: ConfigSequence): (Set[System], List[Step.Step]) =
-    sequenceConfig.getAllSteps.toList.map(Step.step).unzip match {
-      case (s, l) => (s.suml, l)
-    }
+  def read(oid: SPObservationID): SeqexecFailure \/ ConfigSequence =
+      SeqExecService.client(loc).sequence(oid).leftMap(SeqexecFailure.ODBSeqError(_))
+
+  def sequence(sequenceConfig: ConfigSequence):  SeqexecFailure \/ (Set[System], List[Step.Step]) = {
+    val a = sequenceConfig.getAllSteps.toList.map(Step.step).sequenceU
+    a map { _.unzip match {
+      case (l, s) => (l.suml, s)
+    }}
+  }
 
   // todo: it is an error to run a sequence with an existing ExecState != s
   private def runSeq(id: SPObservationID, s: ExecState): Task[(ExecState, NonEmptyList[SeqexecFailure] \/ Unit)] =
     cancelRef.modify(_ - id) *> recordState(id)(s) *> run(go(id), recordState(id))(s)
 
-  def start(id: SPObservationID, sequenceConfig: ConfigSequence): Task[(ExecState, NonEmptyList[SeqexecFailure] \/ Unit)] = {
-    sequence(sequenceConfig) match {
+  def start(id: SPObservationID): SeqexecFailure \/ Unit = for {
+    a <- read(id)
+    b <- start(id, a)
+  } yield b.runAsync(_ => ())
+
+  private def start(id: SPObservationID, sequenceConfig: ConfigSequence): SeqexecFailure \/ Task[(ExecState, NonEmptyList[SeqexecFailure] \/ Unit)] = {
+    sequence(sequenceConfig) map {
       case (s, l) =>
         // TODO: We have the set of systems used by the sequence. Check that they are available.
         runSeq(id, ExecState.initial(l))
     }
   }
 
-  def continue(id: SPObservationID): Task[(Option[ExecState], NonEmptyList[SeqexecFailure] \/ Unit)] =
-    getState(id) >>= {
-      case -\/(e) => Task.now((None, NonEmptyList(e).left))
-      case \/-(s) => runSeq(id, s).map(_.leftMap(_.some))
-    }
+  def continue(id: SPObservationID): SeqexecFailure \/ Unit = continueTask(id) map (_.runAsync(???))
+  private def continueTask(id: SPObservationID): SeqexecFailure \/ Task[(Option[ExecState], NonEmptyList[SeqexecFailure] \/ Unit)] =
+    getState(id).run map {runSeq(id, _).map(_.leftMap(_.some))}
+
+  def state(id: SPObservationID): SeqexecFailure \/ ExecState = getState(id).run
```

Comment:
I guess you can poll this method to follow the sequence, but I'd like to have a method that pushes me changes, it could be like a callback or perhaps other nicer construct